### DISCUSSION TITLE
Does mermaid-js run completely locally/offline?

### DISCUSSION THREAD

**Discussion by MarqueIV** [mermaid-js/mermaid#4412] [Q&A] [answered]
Title sort of says it all.  Wondering if mermaid.js can run completely locally, or if it has to do anything online (e.g. phoning home, etc.) If so, that's a deal-breaker for our use-case which would be an absolute shame, because I absolutely fell in love with it when I came across it's integration in Typora.  Just can't risk information unexpectedly leaking out.

So, can anyone comment if it run completely locally/off-line in our own node.js app (like say a self-authored, internal-use-only VS Code extension)? 

Granted, we'll probably 'vet' the code anyway, but hoping someone can short-circuit that task, saving us the effort if they already know it does use any sort of connectivity.

**Comment by Yokozuna59** [comment, marked as answer]
> So, can anyone comment if it run completely locally/off-line in our own node.js app (like say a self-authored, internal-use-only VS Code extension)?

I guess it does work offline. I'm using this VS Code extenstion, [Markdown Preview Mermaid Support](https://marketplace.visualstudio.com/items?itemName=bierner.markdown-mermaid), to preview locally without the internet. So I guess it will run locally, even if executed from source code or as a package.

**Reply by MarqueIV** [reply]
Phew.  I mean that's what I thought, but you never know.  The good thing is it too is open-source so I can easily skim through the code to see if there's any 'phone home' stuff going on in there.  Thanks for the input/answer! 👍

**Reply by Yokozuna59** [reply]
Anytime! Please mark the discussion as answered if it was the answer you wanted.

**Comment by sidharthv96** [comment]
There's no telemetry in the core mermaid project.

The live editor has telemetry regarding useage (mainly the type of diagram rendered), which uses plausible and the whole data is open for everyone to see at https://plausible.io/mermaid.live. (It's offline now as we haven't paid for it).
